tests/test_wheels.py

```diff
@@ -0,0 +1,88 @@
+import io
+import textwrap
+
+import pytest
+
+from installer.wheels import SuperfulousRecordColumnsWarning, parse_record_file
+
+
+def test_parse_wheel_record_simple():
+    record_content = textwrap.dedent(
+        """\
+        file.py,sha256=AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI,3144
+        distribution-1.0.dist-info/RECORD,,
+        """
+    )
+    records = list(parse_record_file(io.StringIO(record_content)))
+    assert len(records) == 2
+
+    r0 = records[0]
+    assert r0.path.as_posix() == "file.py"
+    assert r0.hash_.name == "sha256"
+    assert r0.hash_.value == "AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI"
+    assert r0.size == 3144
+
+    r1 = records[1]
+    assert r1.path.as_posix() == "distribution-1.0.dist-info/RECORD"
+    assert r1.hash_ is None
+    assert r1.size is None
+
+
+def test_parse_wheel_record_drop_superfulous():
+    record_content = textwrap.dedent(
+        """\
+        file.py,sha256=AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI,3144,
+        distribution-1.0.dist-info/RECORD,,
+        """
+    )
+    record_io = io.StringIO(record_content)
+
+    with pytest.warns(SuperfulousRecordColumnsWarning) as ws:
+        records = list(parse_record_file(record_io))
```

Comment:
Is the expectation to warning once in total, or once per extraneous column? This test doesn't suggest which is expected.

Reply:
Good point, I’ll add more errornous rows to the input. The expectation is to warn once for each row (for no particular reasons beyond it’s what pip does right now and easy to write.)